src/main/index.ts

```diff
@@ -187,6 +200,57 @@ app.whenReady().then(async () => {
   }
 
   SPLASH_SCREEN_WINDOW = createSplashWindow(splashScreenType);
+  SPLASH_SCREEN_WINDOW.on('closed', () => {
+    // We need to drop the variable here to be able to distinguish
+    // in other places whether the splah screen window is still open
+    // or not.
+    SPLASH_SCREEN_WINDOW = undefined;
+  });
+
+  if (KANGAROO_CONFIG.systray) {
+    const systray = new Tray(SYSTRAY_ICON_PATH);
+    systray.setToolTip(KANGAROO_CONFIG.productName);
+
+    const contextMenu = Menu.buildFromTemplate([
+      {
+        label: 'Open',
+        type: 'normal',
+        click() {
+          if (SPLASH_SCREEN_WINDOW) {
+            SPLASH_SCREEN_WINDOW.show();
+          } else if(MAIN_WINDOW) {
+            MAIN_WINDOW.show();
+          }
+        },
+      },
+      {
+        label: 'Restart',
+        type: 'normal',
+        click() {
+          const options: Electron.RelaunchOptions = {
+            args: process.argv,
+          };
+          // https://github.com/electron-userland/electron-builder/issues/1727#issuecomment-769896927
+          if (process.env.APPIMAGE) {
+            console.log('process.execPath: ', process.execPath);
+            options.args?.unshift('--appimage-extract-and-run');
+            options.execPath = process.env.APPIMAGE;
+          }
+          app.relaunch(options);
+          app.quit();
```

Comment:
Looks like relaunch does not include quitting the app?!

Reply:
Yes indeed, the electron API is pretty confusing here ([see docs](https://www.electronjs.org/docs/latest/api/app#apprelaunchoptions)).